lookout/style/format/postprocess.py

```diff
@@ -1,181 +1,265 @@
             if not errors:
-                parsed_after, errors = parse_uast(stub, content_after, filename="",
-                                                  language=feature_extractor.language)
-                if check_uasts_are_equal(parsed_before, parsed_after):
-                    safe_preds.extend((i, i + 1))
-            continue
-        if gt == pred:
-            safe_preds.append(i)
-            continue
-        pred_string = feature_extractor.label_to_str(pred)
-        parsed_before = _parse_code(vnode_parents[id(vn_y)], file_content, stub, parsing_cache,
-                                    feature_extractor.language, node_parents, vn_y.path)
-        if parsed_before is None:
-            continue
-        parent_before, start, end = parsed_before
-        # when the input node value is NOOP i.e. an empty string, the replacement is restricted
-        # to the first occurence
-        output_pred = "".join(n.value for n in vnodes[cur_i:cur_i+2]).replace(vn_y.value,
-                                                                              pred_string, 1)
-        diff_pred_offset = len(pred_string) - len(vn_y.value)
-        try:
-            # to handle mixed indentations, we include the `VirtualNode` following the predicted
-            # one in the output predicted string, and start the rest of the sequence one
-            # `VirtualNode` further to avoid its repetitions
-            start_next_vnode = vn_y.start.offset + len(vn_y.value) + len(vnodes[cur_i + 1].value)
-            content_after = (file_content[:vn_y.start.offset]
-                             + output_pred
-                             + file_content[start_next_vnode:])
-        # in case the prediction to check corresponds to the last label of a file
-        except IndexError:
-            content_after = file_content[:vn_y.start.offset] \
-                + output_pred
-        content_after = content_after[start:end + diff_pred_offset]
-        parent_after, errors_after = parse_uast(
-            stub, content_after, filename="", language=feature_extractor.language)
-        if not errors_after:
-            if check_uasts_are_equal(parent_before, parent_after):
-                safe_preds.append(i)
-    _log.info("Non UAST breaking predictions: %d selected out of %d",
-              len(safe_preds), y_pred.shape[0])
-    vnodes_y = [vn for i, vn in enumerate(list(vnodes_y)) if i in safe_preds]
-    return y[safe_preds], y_pred[safe_preds], vnodes_y, rule_winners[safe_preds], safe_preds
+                result = uast, start, end
+                break
+            current_ancestor = node_parents.get(id(current_ancestor), None)
+        else:
+            result = None
+            self._log.warning("skipped file %s, due to errors in parsing the whole content", path)
+        for descendant in descendants:
+            self._parsing_cache[id(descendant)] = result
+        return result
+
+    def check_file(
```

Comment:
We should keep this method private (`_check_file`) and use `[file]` as the `files` argument of `check` and keep the API simple, at least we should provide a nice docstring explaning the difference between the two (atm it's the same docstring so diving into the code is required).

Reply:
Good point, I think I found a good solution. See new changes.